plugins/apache-airflow/src/components/DagTableComponent/DagTableComponent.tsx

```diff
@@ -146,17 +214,25 @@ export const DagTableComponent = (props: DagTableComponentProps) => {
     return await apiClient.listDags();
   }, []);
 
+  const data = value?.map(el => ({
+    ...el,
+    id: el.dag_id, // table records require `id` attribute
+    dagUrl: `${apiClient.baseUrl}dag_details?dag_id=${el.dag_id}`, // construct path to DAG using `baseUrl`
+  }));
```

Comment:
Shouldn't you be doing this work inside the useAsync above instead? Then the returned value would already be exactly what you need. And then the useEffect below could use just `[data]` as its dependency and not need the react-hooks/exhaustive-deps exclusion.

But then again - now, dagsData and setDagsData feels like they would be entirely unnecessary now, right? I mean, if what comes out of the useAsync is exactly what you need, there's no need to double-store it again in a separate useState holder. Could just use `value` directly instead.

Reply:
I put the map function into useAsync block, but the dagsData and setDagsData  are necessary because we need to update it when click the Paused switch